assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6

```diff
@@ -3,7 +3,7 @@ import { ajax } from 'discourse/lib/ajax';
 
 export default {
   shouldRender(args, component) {
-    const needsButton = component.currentUser && component.currentUser.get('staff');
+    const needsButton = component.currentUser && component.currentUser.get('staff') && component.siteSettings.assign_enabled;
```

Comment:
Wow, this solves 90% of the problem.

Still:

- `api.addUserMenuGlyph`
- `api.addTagsHtmlCallback`
- `api.decorateWidget('post-contents:after-cooked')`

run and add stuff to the interface.